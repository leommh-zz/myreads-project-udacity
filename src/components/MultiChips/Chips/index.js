import React from 'react';
import { Chip, Grid } from '@material-ui/core';

export const MapChips = ({ items, style }) => (
    //Tests whether items exist
    items ? (
        <Grid container direction="row" spacing={8}>
            {items.map( item => (
                <Grid item key={item}>
                    <Chip label={item} style={style.chip}/>
                </Grid>
            ))}
        </Grid>
    ) : ( 
        false 
    )

)

export const ItemChip = ({ item, style }) => (
    //Tests whether items exist
    item ? (
        <Grid>
            <Chip label={item} style={style.chip}/>
        </Grid>
    ) : (
        <Grid>
            <Chip label='None' style={style.chip}/>
        </Grid>
    )
)


