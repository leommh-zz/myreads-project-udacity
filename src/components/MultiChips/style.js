const style = {
  typography: {
    color: "#c8a3ff"
  },
  chip: {
    background: "#bfd3ff"
  }
};

export default style;
