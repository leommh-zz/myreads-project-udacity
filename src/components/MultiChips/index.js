import React from 'react';
import { Typography, Grid } from '@material-ui/core';
import {MapChips} from './Chips';
import If from '../If';
import Style from './Style';

const MultiChips = ({ items, name }) =>{
    return(
        // Tests whether items exist
        <If test={items}> 
            <Grid container item>
                <Typography variant="subtitle1" style={Style.typography}> 
                    {name}
                </Typography> 
                <MapChips items={items} Style={Style}/> 
            </Grid>
        </If>
    )
}

export default MultiChips;
