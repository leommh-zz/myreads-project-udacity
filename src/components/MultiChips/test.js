import React from 'react';
import { shallow } from 'enzyme';
import MultiChips from './index';

describe('[Component] MultiChips', () => {
  
    const items = [
        "item1", 
        "item2"
    ];

    const name = "title1";
    

    it('Display MultiChips Component with props', () => {
        const wrapper = shallow(<MultiChips items={items} name={name} />);
        expect(wrapper).toMatchSnapshot();
    });
    
});