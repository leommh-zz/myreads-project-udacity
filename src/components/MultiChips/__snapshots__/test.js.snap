// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] MultiChips Display MultiChips Component with props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MultiChips
    items={
        Array [
            "item1",
            "item2",
          ]
    }
    name="title1"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <WithStyles(Grid)
        container={true}
        item={true}
>
        <WithStyles(Typography)
                style={
                        Object {
                                "color": "#c8a3ff",
                              }
                }
                variant="subtitle1"
        >
                title1
        </WithStyles(Typography)>
        <Unknown
                Style={
                        Object {
                                "chip": Object {
                                  "background": "#bfd3ff",
                                },
                                "typography": Object {
                                  "color": "#c8a3ff",
                                },
                              }
                }
                items={
                        Array [
                                "item1",
                                "item2",
                              ]
                }
        />
</WithStyles(Grid)>,
      "test": Array [
        "item1",
        "item2",
      ],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <WithStyles(Typography)
            style={
                        Object {
                                    "color": "#c8a3ff",
                                  }
            }
            variant="subtitle1"
>
            title1
</WithStyles(Typography)>,
          <Unknown
            Style={
                        Object {
                                    "chip": Object {
                                      "background": "#bfd3ff",
                                    },
                                    "typography": Object {
                                      "color": "#c8a3ff",
                                    },
                                  }
            }
            items={
                        Array [
                                    "item1",
                                    "item2",
                                  ]
            }
/>,
        ],
        "container": true,
        "item": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "title1",
            "style": Object {
              "color": "#c8a3ff",
            },
            "variant": "subtitle1",
          },
          "ref": null,
          "rendered": "title1",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "Style": Object {
              "chip": Object {
                "background": "#bfd3ff",
              },
              "typography": Object {
                "color": "#c8a3ff",
              },
            },
            "items": Array [
              "item1",
              "item2",
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <WithStyles(Grid)
          container={true}
          item={true}
>
          <WithStyles(Typography)
                    style={
                              Object {
                                        "color": "#c8a3ff",
                                      }
                    }
                    variant="subtitle1"
          >
                    title1
          </WithStyles(Typography)>
          <Unknown
                    Style={
                              Object {
                                        "chip": Object {
                                          "background": "#bfd3ff",
                                        },
                                        "typography": Object {
                                          "color": "#c8a3ff",
                                        },
                                      }
                    }
                    items={
                              Array [
                                        "item1",
                                        "item2",
                                      ]
                    }
          />
</WithStyles(Grid)>,
        "test": Array [
          "item1",
          "item2",
        ],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(Typography)
              style={
                            Object {
                                          "color": "#c8a3ff",
                                        }
              }
              variant="subtitle1"
>
              title1
</WithStyles(Typography)>,
            <Unknown
              Style={
                            Object {
                                          "chip": Object {
                                            "background": "#bfd3ff",
                                          },
                                          "typography": Object {
                                            "color": "#c8a3ff",
                                          },
                                        }
              }
              items={
                            Array [
                                          "item1",
                                          "item2",
                                        ]
              }
/>,
          ],
          "container": true,
          "item": true,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "title1",
              "style": Object {
                "color": "#c8a3ff",
              },
              "variant": "subtitle1",
            },
            "ref": null,
            "rendered": "title1",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "Style": Object {
                "chip": Object {
                  "background": "#bfd3ff",
                },
                "typography": Object {
                  "color": "#c8a3ff",
                },
              },
              "items": Array [
                "item1",
                "item2",
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
